Remove source support for node < 4

zbjornson opened the pull request:
*Branched from https://github.com/Automattic/node-canvas/pull/917 -- please review that PR first*

* Removes `nextTick` bits in stream implementations, per #740 
* Removes preprocessor directives targeting old versions of node.

LinusU commented:
I have merged #917 :tada:, it would be awesome if you could rebase on the latest master :)

zbjornson commented:
@LinusU sorry for the delay. Just rebased, and added a 3rd commit to fix the merge error pointed out by 
@AaronAsAChimp https://github.com/Automattic/node-canvas/pull/740#commitcomment-22033227

LinusU commented:
Nice, thanks!
